glean-core/csharp/Glean/Net/BaseUploader.cs

```diff
@@ -68,7 +80,7 @@ internal static (string, string)[] GetHeadersFromJSONString(string documentId, s
             }
             catch (JsonException)
             {
-                Console.WriteLine($"Error while parsing headers for ping {documentId}");
+                Log.Error("Error while parsing headers for ping {0}", documentId);
```

Comment:
[According to Serilog's docs](https://github.com/serilog/serilog/wiki/Writing-Log-Events) using "templates" is preferred over "messages". I'm not entirely sure that string interpolation gets treated as a template so I changed this to be in line with the other places we format in values.

Reply:
```suggestion
                Log.Error("Error while parsing headers for ping {documentId}");
```

Can this form be used instead here and in other places?

Reply:
Looks like [c# string interpoloation](https://docs.microsoft.com/en-us/dotnet/csharp/tutorials/string-interpolation) is built on top of composite formatting, so this may not be a performance issue with the way Serilog caches templates after all, so yes, we can use the string interpoloation formatting here and other places.  This is cool because I much prefer the readability of that over composite.